### PR TITLE
Skip ConformanceGatewayAPI and ConformanceIngress test when not needed

### PR DESCRIPTION
**Pull request by christarazi**
- .github/workflows: Skip ConformanceGatewayAPI test when not needed
- .github/workflows: Skip ConformanceIngress test when not needed

### CODE REVIEW

File: .github/workflows/conformance-gateway-api.yaml [hunk 1]
```diff
@@ -31,8 +39,92 @@ env:
...
+        sha: ${{ steps.vars.outputs.sha }}
+        context: ${{ github.workflow }}
+        description: GatewayAPI connectivity test in progress...
+        state: pending
+        target_url: ${{ env.check_url }}
+
+  skip-test-run:
+    # If the modified files are not relevant for this test then we can skip
+    # this test and mark it as successful.
+    if: github.event.comment.body == '/test' && needs.check_changes.outputs.tested == 'false'
```
Comment (christarazi) [inline .github/workflows/conformance-gateway-api.yaml]: Hm, maybe this if condition is wrong because this workflow is not triggered on `/test`?

Comment (christarazi) [reply .github/workflows/conformance-gateway-api.yaml]: :heavy_check_mark:

File: .github/workflows/conformance-gateway-api.yaml [hunk 2]
```diff
@@ -31,8 +39,92 @@ env:
   timeout: 5m
 
 jobs:
+  check_changes:
+    name: Deduce required tests from code changes
+    if: |
+      github.event_name == 'schedule' ||
```
Comment (brlbil) [inline .github/workflows/conformance-gateway-api.yaml:45]: `github.event_name == 'schedule'` is checked here but neither workflows are triggered with a schedule.
There are two triggers for the workflows `pull_request` and `push`, both of them already have path filters defined.

The question is what is the expected behavior?

**Review by brlbil — CHANGES_REQUESTED** [review]

File: .github/workflows/conformance-gateway-api.yaml (hunk 2 above)
Comment (christarazi) [reply .github/workflows/conformance-gateway-api.yaml:45]: I copy-pasted from `.github/workflows/conformance-clustermesh.yaml`. I'm not skilled at GH Actions so it's possible I'm doing something completely wrong. 

The desired behavior is for ConformanceGatewayAPI and ConformanceIngress to be skipped if there are changes only to `test/` directory. However, they get stuck as required jobs as seen in https://github.com/cilium/cilium/pull/25325.

Comment (brlbil) [reply .github/workflows/conformance-gateway-api.yaml:45]: I think the referenced issue is about not properly reporting the status. I will discuss this with a couple of stakeholders and we can move forward with a better approach, either changing the triggers for the workflows or handling status reporting properly, and will follow up the issue to a resolution. But I think in this form this PR is not very useful. if no objection let's close this one.

**Pull request closed without merging**